db/maintenance/info_test.go

```diff
@@ -0,0 +1,75 @@
+package maintenance
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/Scalingo/cli/utils"
+	"github.com/Scalingo/go-scalingo/v6"
+)
+
+func TestGetNextMaintenanceLocalWindow(t *testing.T) {
+	now := dateOfTheWeekUTC(time.Wednesday, 8)
+
+	tests := map[string]struct {
+		maintenanceTimeData        time.Time
+		expectedMaintenanceRunTime func(time.Time, scalingo.MaintenanceStatus) time.Time
+	}{
+		"when the maintenance should be executed next week": {
+			maintenanceTimeData: dateOfTheWeekUTC(time.Sunday, 0),
+			expectedMaintenanceRunTime: func(maintenanceData time.Time, _ scalingo.MaintenanceStatus) time.Time {
+				return maintenanceData.AddDate(0, 0, 7)
+			},
+		},
+		"when the maintenance window is ongoing": {
+			maintenanceTimeData: dateOfTheWeekUTC(time.Wednesday, 7),
+			expectedMaintenanceRunTime: func(maintenanceData time.Time, maintenanceType scalingo.MaintenanceStatus) time.Time {
+				if maintenanceType == scalingo.MaintenanceStatusScheduled {
+					return maintenanceData.AddDate(0, 0, 7)
+				}
+				return maintenanceData
+			},
+		},
+		"when the maintenance window is this week": {
+			maintenanceTimeData: dateOfTheWeekUTC(time.Saturday, 7),
+			expectedMaintenanceRunTime: func(maintenanceData time.Time, maintenanceType scalingo.MaintenanceStatus) time.Time {
+				return maintenanceData
+			},
+		},
+	}
+	for _, maintenanceType := range []scalingo.MaintenanceStatus{
+		scalingo.MaintenanceStatusNotified,
+		scalingo.MaintenanceStatusQueued,
+		scalingo.MaintenanceStatusScheduled,
+	} {
+		for testName, test := range tests {
+			t.Run("when the maintenance is "+string(maintenanceType)+"/"+testName, func(t *testing.T) {
```

Comment:
**issue(blocking):** when running your tests, it will display "when the maintenance is when the maintenance window is this week"

Reply:
It actually displays: 
```
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/when_the_maintenance_should_be_executed_next_week
```
But I agree it is not really informative and not a correct syntax.

Reply:
IMO something like the following is better, WDYT:
```
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/it_should_be_executed_this_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/it_should_be_executed_next_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/it_should_be_ongoing
```

Reply:
i read too fast, you are right, i wrongly thought this was just repeating the test name, all good

Reply:
ack FYI, it looks like this:
```
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/it_should_be_executed_this_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/it_should_be_executed_next_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_notified/it_should_be_ongoing
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_queued/it_should_be_executed_next_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_queued/it_should_be_ongoing
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_queued/it_should_be_executed_this_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_scheduled/it_should_be_executed_next_week
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_scheduled/it_should_be_ongoing
TestGetNextMaintenanceLocalWindow/when_the_maintenance_is_scheduled/it_should_be_executed_this_week
```